cli/lib/kontena/cli/master/login_command.rb

```diff
@@ -321,11 +321,21 @@ def select_a_server(name, url)
               exit_with_error "Master #{url} was found from config, but it does not have an URL and no URL was provided on command line"
             end
             return name_match
-          else
-            exit_with_error "Can't find a master with name #{name} from configuration"
+          elsif cloud_auth?
+            masters = cloud_client.get('user/masters')
+            if masters.kind_of?(Hash) && masters['data'].kind_of?(Array)
+              cloud_master = masters['data'].find { |master| master['attributes']['name'] == url }
+              if cloud_master
+                return Kontena::Cli::Config::Server.new(
+                  url: cloud_master['attributes']['url'],
+                  name: config.find_server_by(name: cloud_master['attributes']['name']).nil? ? cloud_master['attributes']['name'] : next_name(cloud_master['attributes']['name'])
```

Comment:
Could this just be `name: next_name(cloud_master['attributes']['name'])`? Looks like `next_name` won't start `succ`'ing the name if it's not a duplicate?